client/post-editor/editor-html-toolbar/index.jsx

```diff
@@ -23,12 +34,79 @@ export class EditorHtmlToolbar extends Component {
 	};
 
 	state = {
+		isPinned: false,
+		isScrollable: false,
+		isScrolledFull: false,
 		openTags: [],
 		selectedText: '',
 		showImageDialog: false,
 		showLinkDialog: false,
 	};
 
+	componentDidMount() {
+		this.pinToolbarOnScroll = throttle( this.pinToolbarOnScroll, 50 );
+		this.hideToolbarFadeOnFullScroll = throttle( this.hideToolbarFadeOnFullScroll, 200 );
+
+		window.addEventListener( 'scroll', this.pinToolbarOnScroll );
+		window.addEventListener( 'resize', this.onWindowResize );
+		this.buttons.addEventListener( 'scroll', this.hideToolbarFadeOnFullScroll );
+
+		this.toggleToolbarScrollableOnResize();
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'scroll', this.pinToolbarOnScroll );
+		window.removeEventListener( 'resize', this.onWindowResize );
+		this.buttons.removeEventListener( 'scroll', this.hideToolbarFadeOnFullScroll );
+	}
+
+	bindButtonsRef = div => {
+		this.buttons = div;
+	}
+
+	onWindowResize = () => {
+		throttle( this.disablePinOnSmallScreens, 400 );
```

Comment:
Uhm, I did a quick test and the various events worked as expected.
I'll see if I were imagining things as soon as possible.